src/golang/lib/engine/authenticate.go

```diff
@@ -43,12 +49,46 @@ func AuthenticateLambdaConfig(ctx context.Context, authConf auth.Config) error {
 		lambda_utils.SnowflakeConnectorType,
 	}
 
-	for _, functionType := range functionsToShip {
-		err := lambda_utils.CreateLambdaFunction(functionType, lambdaConf.RoleArn)
-		if err != nil {
-			return errors.Wrap(err, "Unable to Create Lambda Function")
-		}
+	errGroup, _ := errgroup.WithContext(ctx)
+
+	pullImageChannel := make(chan lambda_utils.LambdaFunctionType, MaxConcurrentDownload)
+
+	pushImageChannel := make(chan lambda_utils.LambdaFunctionType, MaxConcurrentUpload)
```

Comment:
I'm typically more of a fan of defining variables right before you need to use them, can we move these definitions down to where they are immediately needed? It keeps the code a little better module-ized.